readthedocs/proxito/views/serve.py

```diff
@@ -274,6 +286,172 @@ def _get_canonical_redirect_type(self, request):
 
         return None
 
+    def get_using_unresolver(self, request):
+        """
+        Resolve the current request using the new proxito implementation.
+
+        This is basically a copy of the get() method,
+        but adapted to make use of the unresolved to extract the current project, version, and file.
+        """
+        unresolved_domain = request.unresolved_domain
+        # TODO: capture this path in the URL.
+        path = request.path_info
+        try:
+            unresolved = unresolver.unresolve_path(
+                unresolved_domain=unresolved_domain,
+                path=path,
+                append_indexhtml=False,
+            )
+        except (
+            VersionNotFoundError,
+            TranslationNotFoundError,
+            InvalidExternalVersionError,
+        ) as exc:
+            # TODO: find a better way to pass this to the middleware.
+            request.path_project_slug = exc.project.slug
```

Comment:
What if "the middleware" is the one that calls `unresolver.unresolve_path()` and inject it in the `request` instead? That way, it would be available for everybody who needs it.

Probably not a clean implementation or good idea, but I just thought about that and wrote it down just in case 🙃

Reply:
That would require moving the exception control logic to the middleware, and we want to do that only in the docs view.